src/libraries/ArrayUtil.sol

```diff
@@ -397,16 +380,24 @@ library ArrayUtil {
 
     function sortByIndexes(int256[] memory x, uint256[] memory z) internal pure returns (int256[] memory y) {
         y = new int256[](x.length);
-        for (uint256 i; i < x.length; i++) {
+        for (uint256 i; i < x.length; ) {
             y[i] = x[z[i]];
+
+            unchecked {
+                ++i;
+            }
         }
     }
 
     function append(bytes32[] memory x, bytes32 e) internal pure returns (bytes32[] memory y) {
         y = new bytes32[](x.length + 1);
         uint256 i;
-        for (i = 0; i < x.length; i++) {
+        for (i = 0; i < x.length; ) {
```

Comment:
@Fletch153 there are a few places that initialize `i = 0` when it's not necessary. Can you go through the methods to make sure we aren't leaking gas?